packages/babel-parser/src/tokenizer/index.js

```diff
@@ -897,6 +897,21 @@ export default class Tokenizer extends LocationParser {
 
     let total = 0;
 
+    if (this.hasPlugin("numericSeparator")) {
+      // Called from readHexChar
+      if (radix === 16) {
+        const unicode = this.input.slice(this.state.pos, this.state.pos + len);
+
+        if (unicode.includes("_")) {
+          const numSeparatorPos = this.input.indexOf("_");
```

Comment:
This doesn't work in case there is another `_` in the input somewhere before the number. e.g.

```js
_;

"\x1_2"
```

It would be better to move this check inside the `for` loop right after, where there already are the other errors about numeric separators.

Reply:
Got it.